### PR TITLE
naive add Poisson Regression in dm/lr.c(don't handle yest equal zero situation 

### PR DESCRIPTION
**Pull request by zengarden**

### CODE REVIEW

**Comment by liuzhiqiangruc** [comment]
hi，可以将wx编程exp(wx),这样就保证yest>0了， 对于eval和grad函数只会做极少量的修改。你有时间可以尝试修改一下。

**Comment by liuzhiqiangruc** [comment]
编程-->变成

File: dm/lr.c [hunk 1]
```diff
@@ -471,3 +471,84 @@ void free_lr(LR * lr){
...
+        if (val) {
+            for (j = 0; j < len[i]; ++j) {
+                yest += val[offs + j] * x[id[offs + j]];
+            }
+        } else {
+            for (j = 0; j < len[i]; ++j) {
+                yest += x[id[offs + j]];
+            }
+        }
+        loss += y[i] * log(yest) - yest;
```
Comment (liuzhiqiangruc) [inline dm/lr.c]: loss += y[i]*yest - exp(yest);

File: dm/lr.c [hunk 2]
```diff
@@ -471,3 +471,84 @@ void free_lr(LR * lr){
...
+    int i = 0, j = 0, offs = 0;
+    memset(g, 0, sizeof(double) * col);
+    for (offs = i = 0; i < row; i++) {
+        yest = 0.0;
+        if (val) {
+            for (j = 0; j < len[i]; j++) {
+                yest += val[offs + j] * x[id[offs + j]];
+            }
+            // if yest == 0,
+            g[i] += y[i] * val[offs + j] / yest - val[offs + j];
```
Comment (liuzhiqiangruc) [inline dm/lr.c]: 这一行变成：g[i] += (y[i] - exp(yest)) * val[offs + j];

File: dm/lr.c [hunk 3]
```diff
@@ -471,3 +471,84 @@ void free_lr(LR * lr){
...
+        } else {
+            for (j = 0; j < len[i]; j++) {
+                yest += x[id[offs + j]];
+            }
+            g[i] += y[i] / yest - 1.;
```
Comment (liuzhiqiangruc) [inline dm/lr.c]: 这一行变成：g[i] += (y[i] - exp(yest));

**Comment by liuzhiqiangruc** [comment]
你帮我检查下看我的这个改动是否正确，就是对线性加权的结果做了exp之后作为泊松分布的参数来最大化目标变量。多谢。
